server/vehicle-persistence.lua

```diff
@@ -121,4 +118,135 @@ AddEventHandler('entityRemoved', function(entity)
         local passenger = passengers[i]
         SetPedIntoVehicle(passenger.ped, veh, passenger.seat)
     end
-end)
\ No newline at end of file
+end)
+
+if persistence == 'full' then return end
+
+local cachedVehicles = {}
+local config = require 'config.server'
+
+---@param plate string
+---@return boolean
+local function isVehicleSpawned(plate)
+    local vehicles = GetGamePool('CVehicle')
+
+    for i = 1, #vehicles do
+        local vehicle = vehicles[i]
+        if qbx.getVehiclePlate(vehicle) == plate then
+            return true
+        end
+    end
+
+    return false
+end
+
+--- Save the vehicle position to the database
+---@param vehicle number
+---@param coords vector3
+---@param heading number
+local function saveVehiclePosition(vehicle, coords, heading)
+    local type = GetVehicleType(vehicle)
+    if type == 'heli' or type == 'plane' then
+        coords = vec3(coords.x, coords.y, coords.z + 1.0)
+    end
+    exports.qbx_vehicles:SaveVehicle(vehicle, {
```

Comment:
For style, add a space above this. Keep certain code blocks separate from each other.